### PR TITLE
unit tests

### PR DESCRIPTION
**Pull request by mogaal**
- Adding GitIgnore
- Adding unit-tests with localstack and KinD

### CODE REVIEW

File: test/deploy_monitoring_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	})
+
+	defer terraform.Destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+
+	// options := k8s.NewKubectlOptions("", "", "opa")
+
+	// service := k8s.GetService(t, options, "opa")
+	// require.Equal(t, service.Name, "opa")
```
Comment (jasonBirchall) [inline test/deploy_monitoring_test.go:23]: Any chance you can either define what these comments are, or just remove them?

File: test/unit-test/main.tf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    sns            = "http://localhost:4566"
+    sqs            = "http://localhost:4566"
+    ssm            = "http://localhost:4566"
+    stepfunctions  = "http://localhost:4566"
+    sts            = "http://localhost:4566"
+  }
+}
+
+module "opa" {
+  source = "github.com/ministryofjustice/cloud-platform-terraform-opa?ref=unit-tests"
```
Comment (jasonBirchall) [inline test/unit-test/main.tf:37]: Are you able to explain the branch reference here?

**Review by jasonBirchall — CHANGES_REQUESTED** [review]
A couple of questions, but looks good overall.

File: test/deploy_monitoring_test.go (hunk 1 above)
Comment (mogaal) [reply test/deploy_monitoring_test.go]: Fixed, thank you

**Review by jasonBirchall — APPROVED** [review]

**Pull request merged**